hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestOzoneClientMultipartUploadWithFSO.java

```diff
@@ -360,6 +360,36 @@ public void testMultipartUploadWithPartsLessThanMinSize() throws Exception {
         () -> completeMultipartUpload(bucket, keyName, uploadID, partsMap));
   }
 
+  @Test
+  public void testMultipartUploadWithDiscardedUnusedPartSize()
+      throws Exception {
+    // Initiate multipart upload
+    String uploadID = initiateMultipartUpload(bucket, keyName, RATIS, ONE);
+    byte[] data = generateData(10000000, (byte) 97);
+
+    // Upload Parts
+    Map<Integer, String> partsMap = new TreeMap<>();
+
+    // Upload part 1 and add it to the partsMap for completing the upload.
+    String partName1 = uploadPart(bucket, keyName, uploadID, 1, data);
+    partsMap.put(1, partName1);
+
+    // Upload part 2 and add it to the partsMap for completing the upload.
+    String partName2 = uploadPart(bucket, keyName, uploadID, 2, data);
+    partsMap.put(2, partName2);
+
+    // Upload part 3 but do not add it to the partsMap for completing the
+    String partName3 = uploadPart(bucket, keyName, uploadID, 3, data);
+
+    completeMultipartUpload(bucket, keyName, uploadID, partsMap);
+
+    // Check the bucket size. Since part number 2 was not added to the partsMap,
```

Comment:
comment, change part number "3" was not added.

Reply:
Thanks for the find!
I have made the changes!
@sumitagrawl